evm/vm/forks/byzantium/headers.py

```diff
@@ -70,12 +70,13 @@ def create_byzantium_header_from_parent(parent_header, **header_params):
 def configure_byzantium_header(vm, **header_params):
     validate_header_params_for_configuration(header_params)
 
-    with vm.block.header.build_copy(**header_params) as header:
-        if 'timestamp' in header_params and vm.block.header.block_number > 0:
-            parent_header = get_parent_header(header, vm.chaindb)
-            header.difficulty = compute_byzantium_difficulty(
+    with vm.block.header.build_changeset(**header_params) as changeset:
+        if 'timestamp' in header_params and changeset.block_number > 0:
+            parent_header = get_parent_header(changeset.commit(), vm.chaindb)
```

Comment:
Ah, interesting, it surprised me to call `commit()` and keep changing `changeset`. Maybe it's the name `commit()` which implied some finality to me. Some alternatives to consider:

- `changeset.frozen()`
- `changeset.rlp_obj()`
- `changeset.copy()`
- `changeset.emit()`
- `changeset.produce()`
- `changeset.hatch()`
- `changeset.stamp()`
- `changeset.sealed()`

_I suppose this comment really goes in pyrlp_

Reply:
- `changeset.freeze()` would work for me.  I'll modify `commit()` to be equivalent to `freeze() -> close()`